v2/api/network/customizations/load_balancer_extension.go

```diff
@@ -43,55 +46,45 @@ func (extension *LoadBalancerExtension) ModifyARMResource(
 	// the hub type has been changed but this extension has not been updated to match
 	var _ conversion.Hub = typedObj
 
-	inboundNatRuleGVK := getInboundNatRuleGVK(obj)
+	resourceID, hasResourceID := genruntime.GetResourceID(obj)
+	if !hasResourceID {
+		// If we don't have an ARM ID yet, we've not been claimed so just return armObj as is
+		return armObj, nil
+	}
 
-	inboundNatRules := &network.LoadBalancersInboundNatRuleList{}
-	matchingFields := client.MatchingFields{".metadata.ownerReferences[0]": string(obj.GetUID())}
-	err := kubeClient.List(ctx, inboundNatRules, matchingFields)
+	apiVersion, err := genruntime.GetAPIVersion(typedObj, kubeClient.Scheme())
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed listing InboundNatRules owned by LoadBalancer %s/%s", obj.GetNamespace(), obj.GetName())
+		return nil, errors.Wrapf(err, "error getting api version for resource %s while getting status", obj.GetName())
 	}
 
-	armInboundNatRules := make([]genruntime.ARMResourceSpec, 0, len(inboundNatRules.Items))
-	for _, inboundNatRule := range inboundNatRules.Items {
-		inboundNatRule := inboundNatRule
-
-		var transformed genruntime.ARMResourceSpec
-		transformed, err = transformToARM(ctx, &inboundNatRule, inboundNatRuleGVK, kubeClient, resolver)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to transform InboundNatRules %s/%s", inboundNatRule.GetNamespace(), inboundNatRule.GetName())
+	// Get the raw resource
+	raw := make(map[string]any)
```

Comment:
Does this need an initial capacity for the allocation?

Reply:
No it's totally overwritten by the HTTP GET payload (note that it's passed as a ptr below to allow for that), and there's no good way to know what the capacity would need to be as we don't know what we're going to get back from Azure.